src/overlays/actors/ovl_Obj_Wind/z_obj_wind.c

```diff
@@ -28,21 +29,144 @@ ActorInit Obj_Wind_InitVars = {
     (ActorFunc)ObjWind_Draw,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80B24480[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 4000, ICHAIN_STOP),
 };
 
-#endif
+void ObjWind_Init(Actor* thisx, PlayState* play) {
+    ObjWind* this = (ObjWind*)thisx;
+    s32 pad;
+    WaterBox* waterbox;
+    f32 ySurface;
 
-extern InitChainEntry D_80B24480[];
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    if (WaterBox_GetSurface1(play, &play->colCtx, this->actor.world.pos.x, this->actor.world.pos.z, &ySurface,
+                             &waterbox) &&
+        (this->actor.world.pos.y < ySurface)) {
+        this->underWater = 1;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Wind/ObjWind_Init.s")
+void ObjWind_Destroy(Actor* thisx, PlayState* play) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Wind/ObjWind_Destroy.s")
+typedef struct sD_80B2448C {
+    s16 unk_0;
+    s16 unk_2;
+    s16 unk_4;
+    s16 unk_6;
+    s16 unk_8;
+} sD_80B2448C;
```

Comment:
```suggestion
typedef struct ObjWindStruct {
    /* 0x0 */ s16 unk_0;
    /* 0x2 */ s16 unk_2;
    /* 0x4 */ s16 unk_4;
    /* 0x6 */ s16 unk_6;
    /* 0x8 */ s16 unk_8;
} ObjWindStruct;
```
This struct could be cleaned up a bit